src/System.Management/src/System/Management/ManagementScope.cs

```diff
@@ -304,332 +303,145 @@ static WmiNetUtilsHelper()
+                LoadDelegate(ref QualifierPut_f, Interop.Kernel32.GetProcAddress(loadLibrary, "QualifierSet_Put")) &&
+                LoadDelegate(ref QualifierDelete_f, Interop.Kernel32.GetProcAddress(loadLibrary, "QualifierSet_Delete")) &&
+                LoadDelegate(ref QualifierGetNames_f, Interop.Kernel32.GetProcAddress(loadLibrary, "QualifierSet_GetNames")) &&
+                LoadDelegate(ref QualifierBeginEnumeration_f, Interop.Kernel32.GetProcAddress(loadLibrary, "QualifierSet_BeginEnumeration")) &&
+                LoadDelegate(ref QualifierNext_f, Interop.Kernel32.GetProcAddress(loadLibrary, "QualifierSet_Next")) &&
+                LoadDelegate(ref QualifierEndEnumeration_f, Interop.Kernel32.GetProcAddress(loadLibrary, "QualifierSet_EndEnumeration")) &&
+                LoadDelegate(ref GetCurrentApartmentType_f, Interop.Kernel32.GetProcAddress(loadLibrary, "GetCurrentApartmentType")) &&
+                LoadDelegate(ref GetDemultiplexedStub_f, Interop.Kernel32.GetProcAddress(loadLibrary, "GetDemultiplexedStub")) &&
+                LoadDelegate(ref CreateInstanceEnumWmi_f, Interop.Kernel32.GetProcAddress(loadLibrary, "CreateInstanceEnumWmi")) &&
+                LoadDelegate(ref CreateClassEnumWmi_f, Interop.Kernel32.GetProcAddress(loadLibrary, "CreateClassEnumWmi")) &&
+                LoadDelegate(ref ExecQueryWmi_f, Interop.Kernel32.GetProcAddress(loadLibrary, "ExecQueryWmi")) &&
+                LoadDelegate(ref ExecNotificationQueryWmi_f, Interop.Kernel32.GetProcAddress(loadLibrary, "ExecNotificationQueryWmi")) &&
+                LoadDelegate(ref PutInstanceWmi_f, Interop.Kernel32.GetProcAddress(loadLibrary, "PutInstanceWmi")) &&
+                LoadDelegate(ref PutClassWmi_f, Interop.Kernel32.GetProcAddress(loadLibrary, "PutClassWmi")) &&
+                LoadDelegate(ref CloneEnumWbemClassObject_f, Interop.Kernel32.GetProcAddress(loadLibrary, "CloneEnumWbemClassObject")) &&
+                LoadDelegate(ref ConnectServerWmi_f, Interop.Kernel32.GetProcAddress(loadLibrary, "ConnectServerWmi")) &&
+                LoadDelegate(ref GetErrorInfo_f, Interop.Kernel32.GetProcAddress(loadLibrary, "GetErrorInfo")) &&
+                LoadDelegate(ref Initialize_f, Interop.Kernel32.GetProcAddress(loadLibrary, "Initialize")))
+            {
+                // All required delegates were loaded.
+                Initialize_f(CompatSwitches.AllowIManagementObjectQI);
             }
+            else
+            {
+                LoadPlatformNotSupportedDelegates(string.Format(SR.PlatformNotSupported_FrameworkUpdatedRequired, wminet_utilsPath));
+            }
+        }
+        static bool LoadDelegate<TDelegate>(ref TDelegate delegate_f, IntPtr procAddr) where TDelegate : class
+        {
+            return procAddr != null &&
+                (delegate_f = Marshal.GetDelegateForFunctionPointer(procAddr, typeof(TDelegate)) as TDelegate) != null;
```

Comment:
You should use the generic `Marshal.GetDelegateForFunctionPointer<T>` overload. It will make the code friendly to precompilation, and saves you from the cast.

Reply:
Done. For my understanding: why it makes easier for precompilation?

Reply:
Because of the precompilation can tell easily by looking at the generic instantiation which interop stub is needed. It much harder (and sometimes impossible) to do when you pass `typeof(T)` around.

We may want to add a rule about this to the https://github.com/dotnet/platform-compat tool.

Reply:
It is a big problem for full AOT environments like .NET Native or CoreRT that do not have fallback to JIT at runtime.

More details are in 
https://docs.microsoft.com/en-us/dotnet/framework/net-native/missinginteropdataexception-class-net-native

Reply:
Thanks for the info! I will see what we can do on platform-compat is this regard.